compiler/semstmts.nim

```diff
@@ -2178,6 +2178,17 @@ proc semProcAux(c: PContext, n: PNode, kind: TSymKind,
 
   if sfBorrow in s.flags and c.config.cmd notin cmdDocLike:
     result[bodyPos] = c.graph.emptyNode
+  
+  if sfVirtual in s.flags:
+    if c.config.backend == backendCpp:
+      let typ = s.typ.sons[1][0] #TODO error if not a ptr or should we allow ref/objects?
```

Comment:
Why not allow objects here?

Reply:
Was an open question and also me trying to fix the scope. Will do it then, shouldnt be too difficult at this point.